src/libraries/System.Security.Cryptography/tests/X509Certificates/RevocationTests/AiaTests.cs

```diff
@@ -54,6 +54,41 @@ public static void EmptyAiaResponseIsIgnored()
             }
         }
 
+        [Theory]
+        [InlineData(AiaResponseKind.Pkcs12, true)]
+        [InlineData(AiaResponseKind.Cert, false)]
+        public static void AiaAcceptsCertTypesAndIgnoresNonCertTypes(AiaResponseKind aiaResponseKind, bool mustIgnore)
+        {
+            CertificateAuthority.BuildPrivatePki(
+                PkiOptions.AllRevocation,
+                out RevocationResponder responder,
+                out CertificateAuthority root,
+                out CertificateAuthority intermediate,
+                out X509Certificate2 endEntity,
+                pkiOptionsInSubject: false,
+                testName: Guid.NewGuid().ToString());
+
+            using (responder)
+            using (root)
+            using (intermediate)
+            using (endEntity)
+            using (X509Certificate2 rootCert = root.CloneIssuerCert())
+            {
+                responder.AiaResponseKind = aiaResponseKind;
+
+                using (ChainHolder holder = new ChainHolder())
+                {
+                    X509Chain chain = holder.Chain;
+                    chain.ChainPolicy.CustomTrustStore.Add(rootCert);
+                    chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+                    chain.ChainPolicy.VerificationTime = endEntity.NotBefore.AddMinutes(1);
+                    chain.ChainPolicy.UrlRetrievalTimeout = DynamicRevocationTests.s_urlRetrievalLimit;
+
+                    Assert.NotEqual(mustIgnore, chain.Build(endEntity));
```

Comment:
Is it not worth checking the flags returned?  I feel like it's expecting PartialChain

Reply:
It is indeed returning PartialChain.
I will defer this one to @vcsjones since I just tweaked the test without fully understanding the surroundings.

Reply:
We could disable revocation checking during chain building, and then just assert the whole status is PartialChain.

I feel like AIA and revocation checking have intertwined relationship on macOS though, so if that doesn't do the right thing, then assert with `HasFlag` that the chain status is PartialFlag.

Reply:
> It is returning RevocationStatusUnknown | PartialChain | OfflineRevocation for Pkcs12, for Cert it returns NoError.

Yeah, that matches my expectations.  When the kind is Cert things succeed (so `chain.Build` returns `true`, which opposes `mustIgnore`).

So the "is this failing for the right reason?" would be to do something like

```C#
if (mustIgnore)
{
    Assert.Contains(X509ChainStatusFlag.PartialChain, chain.ChainStatus.Select(c => c.Status));

    Assert.Contains(X509ChainStatusFlag.PartialChain, chain.ChainElements[0].Select(c => c.Status));
}
```

(written from memory, may not compile)


Might be worth pursuing in a followup PR, in case it gets wonky on Android/et al.